jedi/api/__init__.py

```diff
@@ -708,7 +708,7 @@ class Interpreter(Script):
     :param namespaces: A list of namespace dictionaries such as the one
         returned by :func:`globals` and :func:`locals`.
     """
-    _allow_descriptor_getattr_default = True
+    _allow_descriptor_getattr_default = settings.allow_descriptor_getattr
```

Comment:
I went overly conservative this time and even keeps support for folks who may have been using the unofficial method previously

Reply:
Just remove this attribute. It's impossible for people to set it, because this is calculated at import time (which means you cannot influence it ever. Just remove this line and use `settings.interpreter_allow_descriptor_getattr` where `_allow_descriptor_getattr_default` is used.